addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/internal/discovery/SmappeeDiscoveryService.java

```diff
@@ -0,0 +1,194 @@
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfoActuator;
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfoAppliance;
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfoSensor;
+import org.openhab.binding.smappee.internal.SmappeeServiceLocationInfoSensorChannel;
+import org.openhab.binding.smappee.internal.ThingProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Discovery class for the Smappee.
+ * This will detect the switches in your home.
+ *
+ * @author Niko Tanghe - Initial contribution
+ */
+public class SmappeeDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int SEARCH_TIME = 60;
+
+    private final Logger logger = LoggerFactory.getLogger(SmappeeDiscoveryService.class);
+    private SmappeeService smappeeService;
+    private ThingUID bridgeUID;
+
+    private ScheduledFuture<?> scheduledJob;
+
+    /**
+     * Whether we are currently scanning or not
+     */
+    private boolean _scanning;
+
+    /**
+     * Constructs the discovery class using the thing IDs that smappee can discover
+     * - actuators (plugs)
+     * - detected appliances
+     */
+    public SmappeeDiscoveryService(SmappeeService smappeeService, ThingUID bridgeUID) {
+        super(ImmutableSet.of(SmappeeBindingConstants.THING_TYPE_ACTUATOR,
+                SmappeeBindingConstants.THING_TYPE_APPLIANCE), SEARCH_TIME, false);
+
+        this.smappeeService = smappeeService;
+        this.bridgeUID = bridgeUID;
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SmappeeBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    /**
+     * Activate the discovery service.
+     */
+    public void activate() {
+        super.activate(null);
+    }
+
+    /**
+     * Deactivate the discovery service.
+     */
+    @Override
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    /**
+     * Starts the scan. This discovery will:
+     * <ul>
+     * <li>Call 'get service location info'
+     * (https://smappee.atlassian.net/wiki/spaces/DEVAPI/pages/8552482/Get+Servicelocation+Info)</li>
+     * <li>For each actuator, add a thing in the inbox (if it doesn't exist already)</li>
+     * <li>For each appliance, add a thing in the inbox (if it doesn't exist already)</li>
+     * </ul>
+     * The process will continue until {@link #stopScan()} is called.
+     */
+    @Override
+    protected void startScan() {
+        if (_scanning) {
+            stopScan();
+        }
+
+        // this should be done by base class ???
+        // somehow this is not working, so starting a scheduler instead
+        startAutomaticRefresh();
+
+        _scanning = true;
+    }
+
+    public void startAutomaticRefresh() {
+        scheduledJob = scheduler.scheduleWithFixedDelay(this::scanForNewDevices, 0, 5, TimeUnit.MILLISECONDS);
```

Comment:
This will be executed with a delay of 5 MILLISECONDS...?

Reply:
Changed to minutes